src/server.py

```diff
@@ -27,38 +35,130 @@ def server():
             if len(part) < buffer_length:
                 message_complete = True
         print(message)
-        conn.send(response_ok())
+
+        try:
+            uri = parse_request(message)
+        except HTTPException as exception:
+            msg = response_error(exception)
+        try:
+            file_data_tuple = resolve_uri(uri)
+        except HTTPException as exception:
+            msg = response_error(exception)
+        try:
+            msg = response_ok(file_data_tuple)
+        except HTTPException:
+            msg = (response_error("500 Internal Server Error"))
```

Comment:
notice how each of these three successive `try...except` clauses have nearly identical code in the except part?  Can you think of a way to avoid that repetition?